### PR TITLE
Make media query rules arrays

### PR DESCRIPTION
**Pull request by kaicataldo**
fixes https://github.com/adobe-community/issues/issues/11450

Second commit is so that I wasn't blocked while working on the corresponding Huxley changes.

### CODE REVIEW

File: parser/index.js [hunk 1]
```diff
@@ -6,28 +6,31 @@ const plugin = postcss.plugin('behance-postcss-focss', () => {
...
 
     root.each((node) => {
-      const descriptor = {
-        rules: {}
-      };
+      const descriptor = {};
 
       if (node.type === 'rule') {
-        descriptor.selector = node.selector;
+        descriptor.selector = node.selector.replace(/\n/g, '');
```
Comment (kaicataldo) [inline parser/index.js]: This removes newlines in selectors. Common use case:
```
.foo,
.bar,
.foo li {
  width: yooge
}
```

File: parser/index.js [hunk 2]
```diff
@@ -6,42 +6,45 @@ const plugin = postcss.plugin('behance-postcss-focss', () => {
...
 
-          descriptor.rules[ruleNode.selector] = rules;
+          descriptor.rules.push(ruleDescriptor);
         });
       }
       else {
         return;
       }
 
+      descriptor.selector = descriptor.selector.replace(/\n/g, '');
```
Comment (kaicataldo) [inline parser/index.js]: Remove newlines from selectors. Common use case:
```css
.foo,
.foo li {
  width: yooge
}
```

File: parser/index.js [hunk 3]
```diff
@@ -6,42 +6,45 @@ const plugin = postcss.plugin('behance-postcss-focss', () => {
...
 
-          descriptor.rules[ruleNode.selector] = rules;
+          descriptor.rules.push(ruleDescriptor);
         });
       }
       else {
         return;
       }
 
+      descriptor.selector = descriptor.selector.replace(/\n/g, ' ');
```
Comment (kaicataldo) [inline parser/index.js]: .foo,
.foo li {
  width: yooge
}

File: parser/index.js [hunk 4]
```diff
@@ -6,42 +6,45 @@ const plugin = postcss.plugin('behance-postcss-focss', () => {
...
         });
       }
       else if (node.type === 'atrule' && node.name === 'media') {
         descriptor.selector = `@${node.name} ${node.params}`;
+        descriptor.rules = [];
 
         node.walkRules((ruleNode) => {
-          const rules = {};
+          const ruleDescriptor = {
+            selector: ruleNode.selector.replace(/\n/g, ' '),
```
Comment (kaicataldo) [inline parser/index.js]: .foo,
.foo li {
  width: yooge
}

File: test/parser/index.js [hunk 5]
```diff
@@ -189,9 +245,41 @@ describe('parser', function() {
...
         done();
-      }).catch(() => {
-        done.fail();
+      }).catch((e) => {
+        done.fail(e);
       });
     });
   });
+
+  it('calling toString() with parsed descriptors and data should return processed styles', function(done) {
```
Comment (kaicataldo) [inline test/parser/index.js]: Parsing a file + toString() works!

**Comment by hzoo** [comment]
👍

**Comment by weikinhuang** [comment]
LGTM

**Pull request merged**